client/extensions/woocommerce/components/form-location-select/states.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * External dependencies
+ *
+ * @format
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { bindActionCreators } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { areLocationsLoaded, getStates } from 'woocommerce/state/sites/locations/selectors';
+import { decodeEntities } from 'lib/formatting';
+import { fetchLocations } from 'woocommerce/state/sites/locations/actions';
+import FormLabel from 'components/forms/form-label';
+import FormSelect from 'components/forms/form-select';
+import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
+
+class FormStateSelectFromApi extends Component {
+	static propTypes = {
+		country: PropTypes.string.isRequired,
+		locationsList: PropTypes.arrayOf(
+			PropTypes.shape( {
+				code: PropTypes.string.isRequired,
+				name: PropTypes.string.isRequired,
+			} )
+		),
+		onChange: PropTypes.func.isRequired,
+		siteId: PropTypes.number.isRequired,
+		translate: PropTypes.func.isRequired,
+		value: PropTypes.string.isRequired,
+	};
+
+	componentWillMount() {
+		const { siteId, isLoaded } = this.props;
+
+		if ( siteId && ! isLoaded ) {
+			this.props.fetchLocations( siteId );
+		}
+	}
+
+	componentWillReceiveProps( { siteId } ) {
+		if ( siteId !== this.props.siteId ) {
+			this.props.fetchLocations( siteId );
+		}
+	}
+
+	renderOption = option => {
+		return (
+			<option key={ option.code } value={ option.code }>
+				{ decodeEntities( option.name ) }
+			</option>
+		);
+	};
+
+	renderDisabled = () => {
+		const { translate } = this.props;
+		return (
+			<FormSelect disabled>
+				<option>{ translate( 'N/A' ) }</option>
+			</FormSelect>
+		);
+	};
+
+	render() {
+		const { locationsList, isLoaded, onChange, translate, value } = this.props;
+		const statesLabel = translate( 'State' );
+
+		return (
+			<div>
+				<FormLabel htmlFor="state">{ statesLabel }</FormLabel>
+				{ isLoaded && ! locationsList.length ? (
+					this.renderDisabled()
+				) : (
+					<FormSelect id="state" name="state" onChange={ onChange } value={ value }>
+						<option key="default" value="" disabled>
+							{ translate( 'Select State' ) }
```

Comment:
Please add translation context - these words have all sorts of multiple meanings (Select as a verb or adjective, State as a political entity or present condition)

Reply:
I can add a comment, but it sounds like `context` should only be used on duplicate strings where the translation might need to be different — [docs](https://github.com/Automattic/i18n-calypso#translate-method).